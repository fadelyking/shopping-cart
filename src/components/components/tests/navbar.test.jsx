import { render, screen } from "@testing-library/react";
import { MemoryRouter, Router } from "react-router-dom";
import NavBar from "../navbar";
import { describe, vi } from "vitest";
import userEvent from "@testing-library/user-event";
import Shop from "../../containers/shop";

describe("Navigation", () => {
	it("Should render the navigation bar", () => {
		const { container } = render(
			<MemoryRouter>
				<NavBar />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});
});

describe("Cart items", async () => {
	it("Add to cart button", async () => {
		const images = [
			{ id: 1, name: "shirt", price: 321, image: "link.com" },
			{ id: 2, name: "pants", price: 22, image: "link2.com" },
		];
		const cart = [
			{ id: 1, name: "shirt", price: 321, image: "link.com" },
			{ id: 2, name: "pants", price: 22, image: "link2.com" },
		];
		const setItemsNum = vi.fn();
		const setCartItems = vi.fn();
		const user = userEvent.setup();
		const { container } = render(
			<MemoryRouter>
				<Shop
					images={images}
					setItemsNum={setItemsNum}
					cart={cart}
					setCartItems={setCartItems}
				/>
			</MemoryRouter>
		);

		const buttons = screen.getAllByRole("button", { name: "Add to Cart" });
		await user.click(buttons[1]);
		expect(setCartItems).toHaveBeenCalled();
		expect(setItemsNum).toHaveBeenCalled();
		expect(cart.length).toBe(2);
	});
});
