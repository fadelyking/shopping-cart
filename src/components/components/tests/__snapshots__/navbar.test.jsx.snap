// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation > Should render the navigation bar 1`] = `
<div>
  <div
    class="nav-bar"
  >
    <div
      class="nav-links"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            <h1>
              Home
            </h1>
          </a>
        </li>
        <a
          href="/shop"
        >
          <li>
            Shop
          </li>
        </a>
      </ul>
    </div>
    <a
      href="/checkout"
    >
      <div
        class="cart"
      >
        <p>
           Cart Icon
        </p>
      </div>
    </a>
  </div>
</div>
`;
